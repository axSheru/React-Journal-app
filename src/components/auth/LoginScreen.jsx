import React from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import validator from 'validator';
import { startGoogleLogin, startLoginEmailPassword } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';

export const LoginScreen = () => {

    const dispatch = useDispatch();

    const [ formValues, handleInputChange ] = useForm({
        email: 'alexp@test.com',
        password: '12345678'
    });

    const { email, password } = formValues;

    const handleLogin = ( e ) => {
        e.preventDefault();

        // NOTE: Add better form error handling like in register form.
        if ( validator.isEmail( email ) && password.trim().length > 0 ) {
            dispatch( startLoginEmailPassword( email, password ) );
        } else {
            console.log( 'Faltan datos.' );
        }
    };

    const handleGoogleLogin = () => {
        dispatch( startGoogleLogin() );
    };
    
    return (
        <>
            <h3 className='auth__title'>Login</h3>

            <form onSubmit={ handleLogin }>
                <input
                    type="text"
                    placeholder='email'
                    name="email"
                    autoComplete='off'
                    className='auth__input'
                    value={ email }
                    onChange={ handleInputChange }
                />
                <input
                    type="password"
                    placeholder='Password'
                    name="password"
                    className='auth__input'
                    value={ password }
                    onChange={ handleInputChange }
                />
                <button
                    type='submit'
                    className='btn btn-primary btn-block'
                >
                    Login
                </button>
                <hr />
                <div className='auth__social-networks'>
                    <p>Login with social networks.</p>
                    <div 
                        className="google-btn"
                        onClick={ handleGoogleLogin }
                    >
                        <div className="google-icon-wrapper">
                            <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                        </div>
                        <p className="btn-text">
                            <b>Sign in with google</b>
                        </p>
                    </div>
                </div>
                <Link
                    to='/auth/register'
                    className='link'
                >
                    Create new account
                </Link>
            </form>
        </>
    );
}
