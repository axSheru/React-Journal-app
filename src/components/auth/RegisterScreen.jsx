import React from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import validator from 'validator';
import { removeError, setError } from '../../actions/ui';
import { useForm } from '../../hooks/useForm';

export const RegisterScreen = () => {

    const dispatch = useDispatch();

    const [ formValues, handleInputChange ] = useForm({
        name: 'Alex P',
        email: 'alexp@test.com',
        password: '12345678',
        password2: '12345678'
    });

    const { name, email, password, password2 } = formValues;

    const handleRegister = ( e ) => {
        e.preventDefault();

        console.log('handleRegister')
        
        if ( isFormValid() ) {
            console.log('Formulario correcto');
        }
    };

    const isFormValid = () => {

        if ( name.trim().length === 0 ) {
            dispatch( setError( 'Name is required.' ) );
            console.log(1);
            return false;
        }

        if ( ! validator.isEmail( email ) ) {
            dispatch( setError( 'Email not valid.' ) );
            console.log(2);
            return false;
        }

        if ( password !== password2 || password.length < 5 ) {
            dispatch( setError( 'Password should be at least 6 characters and both fields should match.' ) );
            console.log(3);
            return false;
        }

        dispatch( removeError() );

        return true;
    };

    return (
        <>
            <h3 className='auth__title'>Register</h3>

            <form>

                <div className="auth__alert-error">
                    Hola mundo
                </div>

                <input
                    type="text"
                    placeholder='Name'
                    name="name"
                    autoComplete='off'
                    className='auth__input'
                    value={ name }
                    onChange={ handleInputChange }
                />
                <input
                    type="text"
                    placeholder='Email'
                    name="email"
                    autoComplete='off'
                    className='auth__input'
                    value={ email }
                    onChange={ handleInputChange }
                />
                <input
                    type="password"
                    placeholder='Password'
                    name="password"
                    className='auth__input'
                    value={ password }
                    onChange={ handleInputChange }
                />
                <input
                    type="password"
                    placeholder='Confirm password'
                    name="password2"
                    className='auth__input'
                    value={ password2 }
                    onChange={ handleInputChange }
                />
                <button
                    type='submit'
                    className='btn btn-primary btn-block mb-5'
                    onClick={ handleRegister }
                >
                    Register
                </button>
                <Link
                    to='/auth/login'
                    className='link'
                >
                    Already registered?
                </Link>
            </form>
        </>
    );
}
