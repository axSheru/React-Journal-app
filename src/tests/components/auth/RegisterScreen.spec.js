import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { RegisterScreen } from '../../../components/auth/RegisterScreen';

describe('Pruebas en el componente RegisterScreen.', () => {

    const middlewares = [ thunk ];
    const mockStore = configureStore(middlewares);

    const initState = {
        ui: {
            loading: false,
            msgError: null
        }
    };

    let store = mockStore( initState );

    store.dispatch = jest.fn();

    const wrapper = mount(
        <Provider store={ store }>
            <MemoryRouter>
                <RegisterScreen />
            </MemoryRouter>
        </Provider>
    );

    test('Debe de hacer match con el snapshot.', () => {

        expect( wrapper ).toMatchSnapshot();

    });
    
});