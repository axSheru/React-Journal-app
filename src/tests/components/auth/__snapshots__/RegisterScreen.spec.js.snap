// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente RegisterScreen. Debe de hacer match con el snapshot. 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Register
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate__faster"
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Alex P"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="alexp@test.com"
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="12345678"
    />
    <input
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value="12345678"
    />
    <button
      className="btn btn-primary btn-block mb-5"
      onClick={[Function]}
      type="submit"
    >
      Register
    </button>
    <a
      className="link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already registered?
    </a>
  </form>,
]
`;
